server/src/main/java/org/elasticsearch/action/admin/indices/template/put/TransportPutComponentTemplateAction.java

```diff
@@ -91,14 +117,36 @@ protected void masterOperation(
         final ActionListener<AcknowledgedResponse> listener
     ) {
         ComponentTemplate componentTemplate = normalizeComponentTemplate(request.componentTemplate(), indexScopedSettings);
-        indexTemplateService.putComponentTemplate(
-            request.cause(),
-            request.create(),
-            request.name(),
-            request.masterNodeTimeout(),
-            componentTemplate,
-            listener
-        );
+        if (componentTemplate.hasDataLifecycle()) {
+            var composableTemplates = indexTemplateService.getTemplatesUsingComponent(state, request.name());
+            var indexPatterns = composableTemplates.values()
+                .stream()
+                .flatMap(it -> it.indexPatterns().stream())
+                .collect(Collectors.toUnmodifiableSet());
+            privilegesCheck.checkCanConfigure(
```

Comment:
I think this is a good place to make the check privilege call (i.e. in the transport action) though we may also push it into the service-level, e.g., inside `putComponentTemplate`.